ext/json/json.c

```diff
@@ -270,6 +288,48 @@ PHP_FUNCTION(json_decode)
 }
 /* }}} */
 
+/* {{{ Validates if a string contains a valid json */
+PHP_FUNCTION(json_validate)
+{
+	char *str;
+	size_t str_len;
+	zend_long depth = PHP_JSON_PARSER_DEFAULT_DEPTH;
+	zend_long options = 0;
+
+	ZEND_PARSE_PARAMETERS_START(1, 3)
+		Z_PARAM_STRING(str, str_len)
+		Z_PARAM_OPTIONAL
+		Z_PARAM_LONG(depth)
+		Z_PARAM_LONG(options)
+	ZEND_PARSE_PARAMETERS_END();
+
+
+	if ((options != 0) && (options != PHP_JSON_INVALID_UTF8_IGNORE)) {
+		zend_argument_value_error(3, "must be a valid flag (allowed flags: JSON_INVALID_UTF8_IGNORE)");
+		RETURN_THROWS();
+	}
+
+	if (!str_len) {
+		JSON_G(error_code) = PHP_JSON_ERROR_SYNTAX;
+		RETURN_FALSE;
+	}
+
+	JSON_G(error_code) = PHP_JSON_ERROR_NONE;
+	
+	if (depth <= 0) {
+		zend_argument_value_error(2, "must be greater than 0");
+		RETURN_THROWS();
+	}
+
+	if (depth > INT_MAX) {
+		zend_argument_value_error(2, "must be less than %d", INT_MAX);
+		RETURN_THROWS();
+	}
+
+	RETURN_BOOL(php_json_validate_ex(str, str_len, options, depth));
```

Comment:
```suggestion
	RETURN_BOOL(php_json_validate_ex(str, str_len, options | PHP_JSON_OBJECT_AS_ARRAY, depth));
```

```
```
	/* For BC reasons, the bool $assoc overrides the long $options bit for PHP_JSON_OBJECT_AS_ARRAY */
	if (!assoc_null) {
		if (assoc) {
			options |=  PHP_JSON_OBJECT_AS_ARRAY;
```

Actually, one other change I missed - this should be passing PHP_JSON_OBJECT_AS_ARRAY as a flag when using it.

Otherwise, php will reject `json_validate({"\u0000anything":true})`

Reply:
It should not matter for `json_validate` as it doesn't set the property so there should be no check for that...

Reply:
Oh, right, that's in php_json_parser_object_update, which the new helper (and php_json_parser_array_update) aren't calling